### PR TITLE
feat: support aten.atan2 converter

### PR DESCRIPTION
**Pull request by chohk88**
# Description

New feature to support aten.atan2 converter. I also add test case including edge case for inputs are zero. 

Fixes # ([issue](https://github.com/pytorch/TensorRT/issues/2586))

## Type of change

Please delete options that are not relevant and/or add your own.
- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by zewenli98 — COMMENTED** [review]
Overall looks good to me. It seems that `get_value_by_condition` is a helper function. Can you move it to `.../dynamo/conversion/converter_utils.py`?

File: py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py [hunk 1]
```diff
@@ -213,6 +216,178 @@ def remainder(
     return fmod2_value
 
 
+def atan2(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    other: TRTTensor,
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:225]: Does this function support `other` being a Python float or does it require `TRTTensor`?

File: py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py [hunk 2]
```diff
@@ -556,3 +731,17 @@ def le(
...
+    condition: TRTTensor,
+) -> TRTTensor:
+    select_layer = ctx.net.add_select(condition, input, other)
+    set_layer_name(select_layer, target, name + "_select", source_ir)
+    return select_layer.get_output(0)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:747]: This can be removed - replace with usage of `impl.condition.select`, from here:
https://github.com/pytorch/TensorRT/blob/76c9ebdc7544f7f666b63e21d13abfc9e2d44c5d/py/torch_tensorrt/dynamo/conversion/impl/condition/ops.py#L102-L113

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:747]: Alternatively, you may need `where`:
https://github.com/pytorch/TensorRT/blob/76c9ebdc7544f7f666b63e21d13abfc9e2d44c5d/py/torch_tensorrt/dynamo/conversion/impl/condition/ops.py#L18-L26

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 3]
```diff
@@ -1391,6 +1391,24 @@ def aten_ops_atanh(
     )
 
 
+@dynamo_tensorrt_converter(torch.ops.aten.atan2.default)
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1394]: If you require one or both input tensors to be `TRTTensor`, consider using:
https://github.com/pytorch/TensorRT/blob/6b8ccbc69f05de3b6275cac4613735934fb559b0/py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py#L62-L66

File: py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py (hunk 1 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py:225]: I think it requires `TRTTensor` because both input and output are `Tensor`

image

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 3 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1394]: I add `enforce_tensor_types` for both input and other.

File: py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py (hunk 2 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/elementwise/ops.py]: I removed `get_value_by_condition` and replaced it with `impl.condition.select.`

**Comment by chohk88** [comment]
@zewenli98 @gs-olive I've made the changes you suggested. I also noticed I missed the case when input or other is a single constant, so I fixed that using `broadcast` function. And, I added test cases to check it's all working right. Thanks for your helpful suggestions!

**Review by narendasan — APPROVED** [review]
LGTM
